### DISCUSSION TITLE
build new temporal dataset

### DISCUSSION THREAD

**Discussion by tliu1023** [pyg-team/pytorch_geometric#4580] [Q&A]
Hi,

I'm new to PyG and I'm a little confused about the usage of it. I'm trying to solve a task like seq2seq prediction but on a series of graphs. That is f(G_t0, G_t1, ... G_tn) -> H_t0, H_t1, ..., H_t0, which is similar to DCRNN but not same. So I'm trying to build a new dataset from scratch using PyG and I'm confused which class should I use, Data or temporalData or Dataset? 

Thank you

**Comment by rusty1s** [comment]
Have you also looked at https://github.com/benedekrozemberczki/pytorch_geometric_temporal? This might be the better tool for what you are trying to achieve.

Otherwise, I don't think `TemporalData` is the appropriate choice here (it expects a list of temporal events rather than graphs coming from different snapshots). As such, you could consider having a `torch_geometric.data.Dataset` that returns a sequence of graphs/`torch_geometric.data.Data` objects within `__getitem__`:
```python
class MyDataset(Dataset):
    ...

def __getitem__(self, idx):
    # get data objects
    return G_t0, G_t1, G_t2


dataset = MyDataset()
loader = DataLoader(dataset, batch_size=8)

for batch_t0, batch_t1, batch_t2 in loader:
    pass
```
